src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -223,6 +267,11 @@ void RadioSpinel<InterfaceType>::Init(bool aResetRadio, bool aSkipRcpCompatibili
     mResetRadioOnStartup = aResetRadio;
 #endif
 
+    VerifyOrDie(aIidList != nullptr, OT_EXIT_INVALID_ARGUMENTS);
+    VerifyOrDie(aIidListLength != 0 && aIidListLength <= OT_ARRAY_LENGTH(mIidList), OT_EXIT_INVALID_ARGUMENTS);
+    mIid = aIidList[0];
+    memcpy(mIidList, aIidList, aIidListLength * sizeof(spinel_iid_t));
```

Comment:
Should we `memset(mIidList, SPINEL_HEADER_INVALID_IID, sizeof(mIidList));` here before `memcpy()` (in addition to doing the same in constructor).

This is just to be safe in case `Init()` is called again with new list that is shorter?

Here we are copying `aIidListLength` and indirectly assume the reset of entries in `mIIdList` are already `SPINEL_HEADER_INVALID_IID`.

Reply:
That's a good point. Made the required change. Thanks.